src/peft/utils/save_and_load.py

```diff
@@ -446,7 +454,11 @@ def renamed_dora_weights(k):
     peft_model_state_dict, mismatched_keys = _find_mismatched_keys(
         model, peft_model_state_dict, ignore_mismatched_sizes=ignore_mismatched_sizes
     )
-    load_result = model.load_state_dict(peft_model_state_dict, strict=False)
+    if init_empty:
+        load_result = model.load_state_dict(peft_model_state_dict, strict=False, assign=True)
```

Comment:
For my own knowledge: 

Can you walk me through a scenario when `set_peft_model_state_dict()` would be required to call with `init_empty=True`?